src/HistoryAPI.ts

```diff
@@ -110,21 +114,64 @@ function getPositions(
 
   debug(query)
 
+  if (format) format = format.toLowerCase()
+
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   v1Client.query(query).then((rows: any[]) => {
-    const resultData = rows.map((row) => {
-      return [row.time.toISOString(), [row.lon, row.lat]]
-    })
-
-    res.json({
-      context,
-      range: {
-        from: from.toString(),
-        to: to.toString(),
-      },
-      values: [{ path: 'navigation.position', method: 'first' }],
-      data: resultData,
-    })
+    if (format && format !== 'json') {
+      // output data in custom format
+      if (supportedFormats.includes(format)) {
+        if (format === 'gpx') {
+          let responseBody = `<?xml version="1.0" encoding="UTF-8" ?>
```

Comment:
How about code like

```
    if (format === 'gpx') {
      outputGpx(rows, res)
    } else if { format === 'json' || format === undefined} {
      outputJson(rows, res)
    } else {
      res.status(400)
      res.json(....error message here....)
    }
```
So create the functions instead of the deep nestings to keep the code tidier.